src/config.rs

```diff
@@ -20,141 +20,129 @@ const PLAIN: u32 = 0;
+            OuterCipherConfig::Twofish => Ok(Box::new(ciphers::TwofishCipher::new(key, iv)?)),
+            OuterCipherConfig::ChaCha20 => {
                 Ok(Box::new(ciphers::ChaCha20Cipher::new_key_iv(key, iv)?))
             }
         }
     }
 
-    pub fn get_iv_size(&self) -> usize {
+    pub(crate) fn get_iv_size(&self) -> usize {
         match self {
-            OuterCipherSuite::AES256 => ciphers::AES256Cipher::iv_size(),
-            OuterCipherSuite::Twofish => ciphers::TwofishCipher::iv_size(),
-            OuterCipherSuite::ChaCha20 => ciphers::ChaCha20Cipher::iv_size(),
-        }
-    }
-
-    pub fn get_key_size(&self) -> usize {
-        match self {
-            OuterCipherSuite::AES256 => ciphers::AES256Cipher::key_size(),
-            OuterCipherSuite::Twofish => ciphers::TwofishCipher::key_size(),
-            OuterCipherSuite::ChaCha20 => ciphers::ChaCha20Cipher::key_size(),
+            OuterCipherConfig::AES256 => ciphers::AES256Cipher::iv_size(),
+            OuterCipherConfig::Twofish => ciphers::TwofishCipher::iv_size(),
+            OuterCipherConfig::ChaCha20 => ciphers::ChaCha20Cipher::iv_size(),
         }
     }
 
     pub(crate) fn dump(&self) -> [u8; 16] {
         match self {
-            OuterCipherSuite::AES256 => CIPHERSUITE_AES256,
-            OuterCipherSuite::Twofish => CIPHERSUITE_TWOFISH,
-            OuterCipherSuite::ChaCha20 => CIPHERSUITE_CHACHA20,
+            OuterCipherConfig::AES256 => CIPHERSUITE_AES256,
+            OuterCipherConfig::Twofish => CIPHERSUITE_TWOFISH,
+            OuterCipherConfig::ChaCha20 => CIPHERSUITE_CHACHA20,
         }
     }
 }
 
-impl TryFrom<&[u8]> for OuterCipherSuite {
-    type Error = OuterCipherSuiteError;
-    fn try_from(v: &[u8]) -> Result<OuterCipherSuite, Self::Error> {
+impl TryFrom<&[u8]> for OuterCipherConfig {
+    type Error = OuterCipherConfigError;
+    fn try_from(v: &[u8]) -> Result<OuterCipherConfig, Self::Error> {
         if v == CIPHERSUITE_AES256 {
-            Ok(OuterCipherSuite::AES256)
+            Ok(OuterCipherConfig::AES256)
         } else if v == CIPHERSUITE_TWOFISH {
-            Ok(OuterCipherSuite::Twofish)
+            Ok(OuterCipherConfig::Twofish)
         } else if v == CIPHERSUITE_CHACHA20 {
-            Ok(OuterCipherSuite::ChaCha20)
+            Ok(OuterCipherConfig::ChaCha20)
         } else {
-            Err(OuterCipherSuiteError::InvalidOuterCipherID { cid: v.to_vec() }.into())
+            Err(OuterCipherConfigError::InvalidOuterCipherID { cid: v.to_vec() }.into())
         }
     }
 }
 
+/// Choices for encrypting protected values inside of databases
 #[derive(Debug, Clone)]
 #[cfg_attr(feature = "serialization", derive(serde::Serialize))]
-pub enum InnerCipherSuite {
+pub enum InnerCipherConfig {
     Plain,
     Salsa20,
     ChaCha20,
 }
 
+/// Errors with inner protected value encryption
 #[derive(Debug, Error)]
-pub enum InnerCipherSuiteError {
+pub enum InnerCipherConfigError {
```

Comment:
@sseemayer same comment as above, could be `InnerCipherError`

Reply:
Since the error is not with the cipher itself, but with its configuration, I would prefer to keep the name as-is.